common/utils/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -18,14 +18,13 @@
 package org.apache.spark
 
 import java.io.FileNotFoundException
+import java.nio.file.FileAlreadyExistsException
 import java.sql.{SQLException, SQLFeatureNotSupportedException}
 import java.time.DateTimeException
 import java.util.ConcurrentModificationException
 
 import scala.collection.JavaConverters._
 
-import org.apache.hadoop.fs.FileAlreadyExistsException
```

Comment:
is this a breaking change?

Reply:
Sort of. IMO it is better to rely on Java's exceptions than on Hadoop's.

Reply:
Had a chat offline and we may choose to keep `SparkFileAlreadyExistsException` inside SparkCore since only server/spark engine side needs this exception. And then we do not need to add `hadoop` dependency on the common-utils thus avoid that stuff on Spark Connect.

Reply:
SGTM